geco/mips/max_cut.py

```diff
@@ -0,0 +1,97 @@
+import itertools
+import random
+
+import networkx as nx
+import pyscipopt as scip
+
+import geco.mips.utilities.naming as naming
+
+
+def tang(n, m, seed=0):
+    """Generates a max-cut instance as described in A.2 in
+        Tang, Y., Agrawal, S., & Faenza, Y. (2019). Reinforcement learning for integer
+        programming: Learning to cut. arXiv preprint arXiv:1906.04859.
+        Args:
+            n (int): number of nodes
+            m (int): number of edges
+            seed (int): seed for randomization
+    """
+    random.seed(seed)
+    graph = nx.generators.gnm_random_graph(n, m)
+    for _, _, data in graph.edges(data=True):
+        data['weight'] = random.randint(0, 10)
+    _, model = naive(graph)
+    return model
+
+
+def empty_edge(graph: nx):
+    model = scip.Model("Odd-Cycle MaxCut")
+
+    edge_variables = {}
+    for u, v, d in graph.edges(data=True):
+        edge_name = naming.undirected_edge_name(u, v)
+        weight = d["weight"]
+        edge_variables[edge_name] = model.addVar(lb=0, ub=1, obj=weight, name=edge_name, vtype='B')
+
+    model.setMaximize()
+
+    return edge_variables, model
+
+
+def naive(graph: nx):
+    model = scip.Model("Naive MaxCut")
+
+    node_variables = {}
+    for v in graph.nodes():
+        node_variables[v] = model.addVar(lb=0, ub=1, obj=0, name=str(v), vtype="B")
+
+    edge_variables = {}
+    for u, v, d in graph.edges(data=True):
+        edge_name = naming.undirected_edge_name(u, v)
+        weight = d["weight"]
+        edge_variables[edge_name] = model.addVar(lb=0, ub=1, obj=weight, name=edge_name, vtype="B")
+
+    model.setMaximize()
+
+    for u, v, d in graph.edges(data=True):
```

Comment:
We could add the following:
If all edge weights are >= 0 only the first two constraints are added (the other two are not needed in this case).